plugins/modules/ntnx_clusters.py

```diff
@@ -0,0 +1,522 @@
+            )
+        ),
+        credentials=dict(
+            type="dict", options=dict(
+                username=dict(type="str"),
+                password=dict(type="str", no_log=True),
+            ),
+        ),
+        proxy_type_list=dict(
+            type="list", elements="str",
+        )
+    )
+
+    smtp_server_spec = dict(
+        email_address=dict(type="str"),
+        server=dict(
+            type="dict", options=dict(
+                name=dict(type="str"),
+                address=dict(
+                    type="dict", options=dict(
+                        ip=dict(type="str"),
+                        ipv6=dict(type="str"),
+                        fqdn=dict(type="str"),
+                        port=dict(type="int"),
+                        is_backup=dict(type="bool"),
+                    ),
+                ),
+                credentials=dict(
+                    type="dict", options=dict(
+                        username=dict(type="str"),
+                        password=dict(type="str", no_log=True),
+                    ),
+                ),
+                proxy_type_list=dict(type="list", elements="str"),
+            )
+        ),
+        server_type=dict(type="str"),
+    )
+
+    http_proxy_whitelist_spec = dict(
+        target=dict(type="str"),
+        target_type=dict(type="str"),
+    )
+
+    default_vswitch_config_spec = dict(
+        nic_teaming_policy=dict(type="str"),
+        uplink_grouping=dict(type="str"),
+    )
+
+    network_spec = dict(
+        external_ip=dict(type="str"),
+        fully_qualified_domain_name=dict(type="str"),
+        external_data_services_ip=dict(type="str"),
+        external_subnet=dict(type="str"),
+        internal_subnet=dict(type="str"),
+        masquerading_ip=dict(type="str"),
+        masquerading_port=dict(type="str"),
+        domain_server=dict(type="dict", options=domain_server_spec),
+        nfs_subnet_whitelist=dict(type="list", elements="str"),
+        name_server_ip_list=dict(type="list", elements="str"),
+        ntp_server_ip_list=dict(type="list", elements="str"),
+        http_proxy_list=dict(type="list", elements="dict", options=http_proxy_list_spec),
+        smtp_server=dict(type="dict", options=smtp_server_spec),
+        http_proxy_whitelist=dict(type="list", elements="dict", options=http_proxy_whitelist_spec),
+        default_vswitch_config=dict(type="dict", options=default_vswitch_config_spec),
+    )
+    authorized_public_key_list_spec = dict(
+        name=dict(type="str"),
+        key=dict(type="str", no_log=False),
+    )
+
+    module_args = dict(
+        cluster=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive, required=True,
+        ),
+        authorized_public_key_list=dict(type="list", elements="dict", options=authorized_public_key_list_spec),
+        timezone=dict(type="str"),
+        supported_information_verbosity=dict(type="str"),
```

Comment:
I beleive this only accepts 4 values. We can put them in choices = []

Reply:
BASIC_PLUS_CORE_DUMP, ALL, BASIC, NOTHING

Reply:
Refer https://www.nutanix.dev/api_references/prism-central-v3/#/90cf8c01b8623-update-a-existing-cluster